backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunServiceImpl.java

```diff
@@ -1779,28 +1817,30 @@ public void findExperimentRuns(
             ModelDBServiceResourceTypes.PROJECT, projectId, ModelDBServiceActions.READ);
       }
 
-      GetResourcesResponseItem entityResource =
-          roleService.getEntityResource(
-              request.getProjectId(), ModelDBServiceResourceTypes.PROJECT);
+      UserInfo currentLoginUserInfo = authService.getCurrentLoginUserInfo();
       ExperimentRunPaginationDTO experimentRunPaginationDTO =
-          experimentRunDAO.findExperimentRuns(
-              projectDAO, authService.getCurrentLoginUserInfo(), request);
+          experimentRunDAO.findExperimentRuns(projectDAO, currentLoginUserInfo, request);
       List<ExperimentRun> experimentRuns = experimentRunPaginationDTO.getExperimentRuns();
       FindExperimentRuns.Response response =
           FindExperimentRuns.Response.newBuilder()
               .addAllExperimentRuns(experimentRuns)
               .setTotalRecords(experimentRunPaginationDTO.getTotalRecords())
               .build();
-      Map<String, Long> auditResourceMap = new HashMap<>();
-      experimentRuns.forEach(
-          experimentRun ->
-              auditResourceMap.put(experimentRun.getId(), entityResource.getWorkspaceId()));
-      saveAuditLog(
-          Optional.empty(),
-          ModelDBServiceActions.READ,
-          auditResourceMap,
-          ModelDBUtils.getStringFromProtoObject(request),
-          ModelDBUtils.getStringFromProtoObject(response));
+      if (experimentRuns.isEmpty()) {
```

Comment:
The workspace of the request doesn't care if you return zero or more. So I'm not sure why we added this conditional.

Similar comment for other locations.

Reply:
We have added those condition because if those entities will be zero then we need the workspace otherwise we will take workspace id from return entities so on that case we do not want to fetch workspace again because it will already filter and returns by workspace.